#
# Submitted-By: Erwin Rol Mar 19 2006
# Committed-By: Erwin Rol
#
# Error:
# Library path for ALSA library
#
# Description:
# Patch taken from Fedora Core 5 RPM
#
# State:
# Unknown
#

Index: SDL-1.2.9/configure.in
===================================================================
--- SDL-1.2.9/configure.in	(revision 58)
+++ SDL-1.2.9/configure.in	(working copy)
@@ -292,12 +292,13 @@
 [  --enable-alsa-shared     dynamically load ALSA audio support [default=yes]],
                           , enable_alsa_shared=yes)
             if test "x`echo $ALSA_LIBS | grep -- -L`" = "x"; then
-                if test "x`ls /lib/libasound.so.* 2> /dev/null`" != "x"; then
-                    ALSA_LIBS="-L/lib $ALSA_LIBS"
-                elif test "x`ls /usr/lib/libasound.so.* 2> /dev/null`" != "x"; then
-                    ALSA_LIBS="-L/usr/lib $ALSA_LIBS"
-                elif test "x`ls /usr/local/lib/libasound.so.* 2> /dev/null`" != "x"; then
-                    ALSA_LIBS="-L/usr/local/lib $ALSA_LIBS"
+	        pfx=${libdir##*/}
+                if test "x`ls /${pfx}/libasound.so.* 2> /dev/null`" != "x"; then
+                    ALSA_LIBS="-L/${pfx} $ALSA_LIBS"
+                elif test "x`ls /usr/${pfx}/libasound.so.* 2> /dev/null`" != "x"; then
+                    ALSA_LIBS="-L/usr/${pfx} $ALSA_LIBS"
+                elif test "x`ls /usr/local/${pfx}/libasound.so.* 2> /dev/null`" != "x"; then
+                    ALSA_LIBS="-L/usr/local/${pfx} $ALSA_LIBS"
                 fi
             fi
             alsa_lib_spec=`echo $ALSA_LIBS | sed 's/.*-L\([[^ ]]*\).*/\1\/libasound.so.*/'`
@@ -425,7 +426,8 @@
                 AC_ARG_ENABLE(arts-shared,
 [  --enable-arts-shared    dynamically load aRts audio support [default=yes]],
                               , enable_arts_shared=yes)
-                arts_lib_spec="$ARTSC_PREFIX/lib/libartsc.so.*"
+	        pfx=${libdir##*/}
+                arts_lib_spec="$ARTSC_PREFIX/${pfx}/libartsc.so.*"
                 arts_lib=`ls $arts_lib_spec | sed 's/.*\/\(.*\)/\1/; q'`
                 echo "-- $arts_lib_spec -> $arts_lib"
                 if test x$use_dlopen != xyes && \
